File: frontend/src/types/importBatch.ts
export interface ImportBatchRow {
  row_key: string;
  source_row: number;
  status: 'ready' | 'invalid' | 'failed' | 'created';
  data: Record<string, string>;
  error?: string | null;
  result?: { record_id?: number; entity?: string; [key: string]: unknown } | null;
}
export interface ImportBatch {
  id: number;
  entity: string;
  filename: string;
  version: number;
  created_at: string;
  updated_at: string;
  total_rows: number;
  counts: Record<string, number>;
  created_records: number;
  rows: ImportBatchRow[];
  row_offset: number;
  has_more_rows: boolean;
  requires_credentials: boolean;
}
export interface ImportBatchHistory {
  batches: ImportBatch[];
  has_more: boolean;
}

File: backend/app/schemas/import_batch.py
from datetime import datetime
from typing import Dict, List, Optional

from pydantic import Field

from app.schemas.base import UTCModel


class ImportBatchRowResponse(UTCModel):
    row_key: str
    source_row: int
    status: str
    data: Dict[str, str]
    error: Optional[str] = None
    result: Optional[dict] = None


class ImportBatchResponse(UTCModel):
    id: int
    entity: str
    filename: str
    version: int
    created_at: datetime
    updated_at: datetime
    total_rows: int
    counts: Dict[str, int]
    created_records: int
    rows: List[ImportBatchRowResponse] = Field(default_factory=list)
    row_offset: int = 0
    has_more_rows: bool = False
    requires_credentials: bool = False


class ImportBatchHistory(UTCModel):
    batches: List[ImportBatchResponse]
    has_more: bool
